adapters/coap-vertx-base/src/main/java/org/eclipse/hono/adapter/coap/DeviceRegistryBasedCertificateVerifier.java

```diff
@@ -100,18 +105,28 @@ public DeviceRegistryBasedCertificateVerifier(
     /**
      * Validate certificate used by a device in a x509 based DTLS handshake and load device.
      *
+     * @param session The DTLS session to be negotiated with the device.
      * @param certPath certificate path.
      * @param span tracing span.
      */
-    private Future<AdditionalInfo> validateCertificateAndLoadDevice(final CertPath certPath, final Span span) {
+    private Future<AdditionalInfo> validateCertificateAndLoadDevice(
+            final DTLSSession session,
```

Comment:
Just to mention: in Californium 3 the dtls-session is replaced by the peer's address and server-names. Should be easy to be migrated then.

Reply:
Does this mean that the `DTLSSession` class will no longer exist or that you are not passing it around anymore?

Reply:
The name still exists, but it contains only the session artifacts (e.g. master secret, session id). The relation's crypto context (association keys, record sequence number) is now in a class called `DTLSContext`. That makes things like a "DTLS session cache" clearer.
 
I don't pass DTLSSession/Context longer around. FMPOV, these classes contains too much information. 
I tried to reduce the data to that part, which is intended for the cases. (In some cases, that may require to adapt the API again, but then with a use-case in mind).

Reply:
Sounds good. My understanding is that by *server-names* you mean the names that have been conveyed in the SNI extension, right?

Reply:
Yes, the names from the SNI extension.